vaadin-crud-flow/src/main/java/com/vaadin/flow/component/crud/Crud.java

```diff
@@ -187,8 +192,12 @@ private void registerHandlers() {
                 ((ComponentEventListener<CancelEvent<E>>) e -> {
                     cancelListeners.forEach(listener -> listener.onComponentEvent(e));
 
-                    setOpened(false);
-                    getEditor().clear();
+                    if ((this.gridActiveItem != null && this.getEditor().getItem() == this.gridActiveItem) || this.gridActiveItem == null) {
+                        setOpened(false);
+                        getEditor().clear();
+                    } else {
+                        getEditor().setItem(this.gridActiveItem);
```

Comment:
This seems a little bit odd to me. Why we are setting the `gridActiveItem` to be edited on `CancelEvent`?

Reply:
Removed if/else to always clean/close

Reply:
Uhmm.. Just came back to me the reason I did this. The scenario is the following (with `editOnClick` enabled):
1. User opens and item to edit.
2. User make changes to the item.
3. User clicks decides to open another item.
4. CRUD will prompt a confirmation dialog to discard changes.
5. User clicks "Discard", which sends a Cancel event.

In that case, I don't want to close the editor, but rather keep it open with the item that is current active.

Reply:
Added a check to prevent the editor to close on such case.